spark/src/main/scala/org/apache/spark/sql/delta/managedcommit/AbstractBatchBackfillingCommitStore.scala

```diff
@@ -119,17 +121,27 @@ trait AbstractBatchBackfillingCommitStore extends CommitStore with Logging {
       fileStatus: FileStatus): Unit = {
     val targetFile = FileNames.deltaFile(logPath(tablePath), version)
     logInfo(s"Backfilling commit ${fileStatus.getPath} to ${targetFile.toString}")
+    val commitContentIterator = logStore.readAsIterator(fileStatus, hadoopConf)
     try {
       logStore.write(
         targetFile,
-        logStore.readAsIterator(fileStatus, hadoopConf),
+        commitContentIterator,
         overwrite = false,
         hadoopConf)
+      registerBackfill(tablePath, version, targetFile)
     } catch {
       case _: FileAlreadyExistsException =>
         logInfo(s"The backfilled file $targetFile already exists.")
+    } finally {
+      commitContentIterator.close()
     }
   }
 
+  /** Callback to tell the CommitStore that all commits <= `untilVersion` are backfilled. */
+  protected[managedcommit] def registerBackfill(
```

Comment:
Why is this `protected[managedcommit]` and not protected like other APIs?

This class could be extended by CommitStores even outside this module.

Reply:
Done